sass-style-guide.md

```diff
@@ -59,87 +42,111 @@ You should order your rules from generic to specific. Example:
-* Be consistent in declaration order, cluster related properties (see above).
-* Be consistant with quotes – we’re using single quotes.
-* Quote attribute values in selectors, e.g. `input[type='checkbox']`.
-* Write one selector per line, one rule per line.
-* Put a space after `:` in property declarations, e.g. `margin: 0`
-* Include a space after each comma in comma-separated property or function values, e.g. `rgba(255, 255, 255, 0)`.
-* Use rgb and rgba color codes, because most of us can envision what color results by mixing red, green and blue.
-
-It could be a quick-win to use [Sass Lint](https://github.com/sasstools/sass-lint) in your projects.
-
-## Coding conventions
-
-### Avoid dangerous selectors
-
-If a selector is too generic, it’s dangerous. In 99% of cases you have to overwrite this rule somewhere. Be more specific. Try using a class instead. (Exception: CSS-Resetstyles)
-
-**bad**
+## Syntax und Formatierung
+
+Eine der einfachsten Formen eines Styleguides ist ein Set von Regeln bezüglich Syntax und Formatierung zu definieren, so dass alle Entwickler eine gemeinsame Basis haben und der Code gut zu lesen ist.
+
+### Schreibweise
+
+- Einrückung: zwei Leerzeichen, keine Tabs
+- Multi-Line-Stil: Ein Selektor pro Zeile, eine Deklaration pro Zeile
+- Selektorverschachtelung: maximal drei Ebenen tief
+- Alle Bezeichnungen werden auf englisch und klein geschrieben. Setzt man mehrere Wörter zu einem Selektor zusammen, trennt diese ein Bindestrich: `.data-table`.
+- Wir verwenden einfache Anführungszeichen: `font-family: 'Andale Mono'`.
+- Werte in Attributselektoren werden in Anführungszeichen gesetzt: `input[type='checkbox']`.
+- Nach einem `:` hinter der CSS-Eigenschaft gehört ein Leerzeichen: `margin: 0`
+- Nach einem `,` in einer kommaseparierten Liste gehört ein Leerzeichen: `rgba(255, 255, 255, 0.5)`
+- Wir verwenden rgb und rgba Farbcodes, weil sich die meisten von uns vorstellen können, welche Farbe durch das Mischen von Rot, Grün und Blau zustande kommt.
+
+### Deklarationsreihenfolge
+
+Um nicht ständig nach einer CSS-Eigenschaft suchen zu müssen, verwenden wir eine feste Deklarationsreihenfolge. Sie ist in logische Blöcke gegliedert, die man sich im Detail in der Linter-Config (TODO: verlinken) ansehen kann.
+
+#### 1. Übergeordnetes
+
+- z.B. content, will-change
+
+#### 2. Allgemeine Sichtbarkeit
+
+- z.B. opacity, visibility, display
+
+#### 3. Layout-Mechanismen
+
+- Flexbox: z.B. justify-content, align-items, flex
+- Grid: z.B. columns, column-gap, column-rule
+- Table: z.B. table-layout, border-collapse, border-spacing
+- List: z.B. list-style, counter-increment, counter-reset
+
+#### 4. Positionierung im Raum
+
+- z.B. float, position, vertical-align
+
+#### 5. Box Model inkl. Transformation
+
+- z.B. box-sizing, width, height, margin, border, padding, transform
```

Comment:
* [ ] Die Beispiele sollten ja auch schon in der Reihenfolge sein wie sie im Detail vorgegeben ist. Und hier würde ich von außen nach innen erwarten: `width, height, padding, border, margin`
* Warum transform hier? Die Prozentangaben sind zwar abhängig von den Ausmaßen der Box. Aber verwendet man das nicht am häufigsten zur Bewegung im Raum und würde daher eher unter 4. passen?